src/test/java/staffconnect/model/person/ModuleTest.java

```diff
@@ -0,0 +1,66 @@
+package staffconnect.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static staffconnect.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class ModuleTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Module(null));
+    }
+
+    @Test
+    public void constructor_invalidModule_throwsIllegalArgumentException() {
+        String invalidModule = "";
+        assertThrows(IllegalArgumentException.class, () -> new Module(invalidModule));
+    }
+
+    @Test
+    public void isValidModule() {
+        // null phone number
+        assertThrows(NullPointerException.class, () -> Module.isValidModule(null));
+
+        // invalid phone numbers
+        assertFalse(Module.isValidModule("")); // empty string
+        assertFalse(Module.isValidModule(" ")); // spaces only
+        assertFalse(Module.isValidModule("ABCD")); // letters only
+        assertFalse(Module.isValidModule("1234")); // numbers only
+        assertFalse(Module.isValidModule("2103T")); // missing prefix
+        assertFalse(Module.isValidModule("C2103T")); // only 1 prefix
+        assertFalse(Module.isValidModule("CS2103TTTT")); // too many suffix
+        assertFalse(Module.isValidModule("CSCSCS2103T")); // too many prefix
+        assertFalse(Module.isValidModule("CS210310101010T")); // too many numbers
+        assertFalse(Module.isValidModule("CS21T")); // too little numbers
+        assertFalse(Module.isValidModule("cs2103t")); // suffix not capitalised
+        assertFalse(Module.isValidModule("cs2103T")); // prefix not capitalised
+
+        // valid phone numbers
```

Comment:
Nits misleading comments in this file, please fix in future commits!

Reply:
will do